friendshipper/src-tauri/src/repo/operations/locks/verify.rs

```diff
@@ -1,21 +1,79 @@
+use std::path::PathBuf;
 use std::sync::Arc;
 
 use anyhow::anyhow;
 use axum::{extract::State, Json};
 
 use ethos_core::types::errors::CoreError;
+use ethos_core::types::locks::Lock;
 use ethos_core::types::locks::VerifyLocksResponse;
 
 use crate::state::AppState;
+use crate::system::unreal::CanUseCommandlet;
+use crate::system::unreal::OFPANameCache;
 
 pub async fn verify_locks_handler(
     State(state): State<Arc<AppState>>,
 ) -> Result<Json<VerifyLocksResponse>, CoreError> {
-    match state.git().verify_locks().await {
-        Ok(output) => Ok(Json(output)),
-        Err(e) => Err(CoreError(anyhow!(
-            "Error executing diff: {}",
-            e.to_string()
-        ))),
+    let mut response_data: VerifyLocksResponse = match state.git().verify_locks().await {
+        Ok(output) => output,
+        Err(e) => {
+            return Err(CoreError(anyhow!(
+                "Error executing diff: {}",
+                e.to_string()
+            )))
+        }
+    };
+
+    let mut ours_paths: Vec<String> = response_data.ours.iter().map(|v| v.path.clone()).collect();
+    let mut theirs_paths: Vec<String> = response_data
+        .theirs
+        .iter()
+        .map(|v| v.path.clone())
+        .collect();
+
+    let ours_len = ours_paths.len();
+    let theirs_len = ours_paths.len();
+
+    let mut combined_paths: Vec<String> = Vec::with_capacity(ours_len + theirs_len);
+    combined_paths.append(&mut ours_paths);
+    combined_paths.append(&mut theirs_paths);
+
+    let repo_path = state.app_config.read().repo_path.clone();
+    let uproject_path = state
+        .app_config
+        .read()
+        .get_uproject_path(&state.repo_config.read());
+    let engine_path = state
+        .app_config
+        .read()
+        .load_engine_path_from_repo(&state.repo_config.read())
+        .unwrap_or_default();
+
+    let ofpa_names = OFPANameCache::get_names(
+        state.ofpa_cache.clone(),
+        &PathBuf::from(repo_path),
+        &uproject_path,
+        &engine_path,
+        &combined_paths,
+        CanUseCommandlet::FallbackOnly,

```

Comment:
This is totally shippable as-is, but this makes me wish for some kind of engine interface that's just like `Engine::sanitize_paths(Vec<String>)`.

I'll think about this as I plot out supporting different engines in f11r.

Reply:
To be fair, it's almost there. We would just need to have another interface in `AppState` that switches between implementations for an engine based on a config setting for the project. Then this code could just live in the unreal version.

Reply:
For sure - this is a pattern we need to figure out for a bunch of different aspects of the tool. For example, we likely want to support non-AWS s3-like backends or Kubernetes clusters that aren't AWS EKS. Support means to me just having an interface someone else can PR implementations of.